features/vaadin-opennms-pluginmanager/src/main/java/org/opennms/features/pluginmgr/vaadin/config/opennms/PluginManagerAdminApplication.java

```diff
@@ -0,0 +1,214 @@
+import java.net.URL;
+import java.util.Map;
+
+import org.opennms.features.pluginmgr.SessionPluginManager;
+import org.opennms.features.pluginmgr.vaadin.config.opennms.internal.HttpServletRequestVaadinImpl;
+import org.opennms.features.pluginmgr.vaadin.pluginmanager.PluginManagerUIMainPanel;
+import org.opennms.web.api.OnmsHeaderProvider;
+
+import com.vaadin.annotations.Theme;
+import com.vaadin.annotations.Title;
+import com.vaadin.server.ExternalResource;
+import com.vaadin.server.Page;
+import com.vaadin.server.VaadinRequest;
+import com.vaadin.server.Page.Styles;
+import com.vaadin.shared.ui.label.ContentMode;
+import com.vaadin.ui.Alignment;
+import com.vaadin.ui.CustomLayout;
+import com.vaadin.ui.HorizontalLayout;
+import com.vaadin.ui.Label;
+import com.vaadin.ui.Link;
+import com.vaadin.ui.UI;
+import com.vaadin.ui.VerticalLayout;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Plugin Manager Administration Application.
+ */
+@Theme("opennms")
+@Title("Plugin Manager Administration")
+@SuppressWarnings("serial")
+public class PluginManagerAdminApplication extends UI {
+	private static final Logger LOG = LoggerFactory.getLogger(PluginManagerAdminApplication.class);
+
+	/*- VaadinEditorProperties={"grid":"RegularGrid,20","showGrid":true,"snapToGrid":true,"snapToObject":true,"movingGuides":false,"snappingDistance":10} */
+
+	//headerLinks map of key= name and value=url for links to be placed in header of page
+    private Map<String,String> headerLinks;
+    
+	private OnmsHeaderProvider m_headerProvider;
+	private String m_headerHtml;
+	private VaadinRequest m_request;
+	private VerticalLayout m_rootLayout;
+
+	private SessionPluginManager sessionPluginManager;
+	
+	public SessionPluginManager getSessionPluginManager() {
+		return sessionPluginManager;
+	}
+
+	public void setSessionPluginManager(SessionPluginManager sessionPluginManager) {
+		this.sessionPluginManager = sessionPluginManager;
+	}
+
+	/**
+	 * headerLinks map of key= name and value=url for links to be placed in header of page
+	 * @return
+	 */
+	public Map<String,String> getHeaderLinks() {
+		return headerLinks;
+	}
+
+	/**
+	 * @param headerLinks map of key= name and value=url for links to be placed in header of page
+	 */
+	public void setHeaderLinks(Map<String,String> headerLinks) {
+		this.headerLinks = headerLinks;
+	}
+
+	// imported ideas from org.opennms.features.vaadin.nodemaps.internal.NodeMapsApplication
+	public OnmsHeaderProvider getHeaderProvider() {
+		return m_headerProvider;
+	}
+
+	public void setHeaderProvider(OnmsHeaderProvider headerProvider) {
+		this.m_headerProvider = headerProvider;
+	}
+
+	public void setHeaderHtml(final String headerHtml) {
+		m_headerHtml = headerHtml;
+	}
+
+	private void addHeader() {
+		if (m_headerProvider != null) {
+			try {
+				URL pageUrl = Page.getCurrent().getLocation().toURL();
+				setHeaderHtml(m_headerProvider.getHeaderHtml(new HttpServletRequestVaadinImpl(m_request, pageUrl)));
```

Comment:
Simply do `((VaadinServletRequest) request).getHttpServletRequest()` instead of wrapping the request.